### DISCUSSION TITLE
What happen when a Kustomize object configured with prune:false being deleted?

### DISCUSSION THREAD

**Discussion by dieend** [fluxcd/flux2#2102] [Q&A] [answered]
We are trying to setup the process on how to clean resources that no longer being used. By default, we are configuring `prune:false` to reduce any unintended mistake. I wonder if deleting the kustomize object directly will still trigger garbage collection?

[Relevant doc](https://fluxcd.io/docs/components/kustomize/kustomization/#garbage-collection):
> Garbage collection means that the Kubernetes objects that were previously applied on the cluster but are missing from the current source revision, are removed from cluster automatically. Garbage collection is also performed when a Kustomization object is deleted, triggering a removal of all Kubernetes objects previously applied on the cluster.

Reading it once seems deleting the kustomization object will still trigger garbage collection, no matter what the prune config is configured to, and you need to explicitly disable pruning for certain resources by either labeling or annotating them with `kustomize.toolkit.fluxcd.io/prune: disabled` .

**Comment by stefanprodan** [comment, marked as answer]
While `prune` is set to `false` Flux will not delete resources, no matter if you delete manifests from Git or you delete the Flux Kustomization from the cluster.

**Reply by mloskot** [reply]
And, after applying that change, is this sufficient verification that resources won't be deleted to just follow this suggestion from https://fluxcd.io/flux/faq/#how-can-i-safely-rename-a-flux-kustomization?

> To make sure that the change has been acknowledged by Flux, run `flux export kustomization <name>` and check that pruning is disabled. 

If `kustomize.toolkit.fluxcd.io/prune: disabled` variant was applied then, I guess, the verification would be to run through all resources in search for that annotation, if applied as expected.
